docs/design/resource-reservation-design.md

```diff
@@ -0,0 +1,91 @@
+priority is more important than waiting time.
+* No matter how many resources high-priority jobs requests and how much time they have already waited for, they should be
+scheduled first.
+* When jobs are at same priority but waiting time differs, job who waits for the longest time should be scheduled first.
+
+## Design
+### Big Job Recognition
+As volcano is a general platform, we tend to support both custom mode and automation mode to recognize Big Jobs.
+#### Custom Mode
+Users can set **request resource** or **waiting duration** as standard. Jobs who request resources more than settings or 
+wait longer than standard line will be treated as potential Big Job. Volcano will choose the Big Job who has the highest 
+priority and above the standard line most as the Big Job.
+#### Automation Mode
+If not config standard line, volcano will order jobs to be scheduled in session by priority and waiting time. The job 
+with the highest priority and waiting for the longest time will be selected as the Big Job.                                                                                
+volcano scheduler will check if there is a Big Job selected in each session. If no Big Job selected, volcano will select
+a Big Job according to the strategy above.
+### Locked Nodes
+As job consists of some tasks and each task corresponds to a pod, scheduler will select a series of nodes who can satisfy
+these pods. These nodes will be locked and no pod can be scheduled to them until the Big Job is scheduled.  
+There are three schemes as follows:
+#### Cluster Lock
+In order to schedule Big Job as soon as possible, lock all nodes in cluster to reserve resource for it. This scheme is 
+suitable for task type with fast throughput. As to long-running task, scheduler performance will be severely degraded.
+#### Multi-Node Lock
+In order to balance scheduler performance and resource reservation, we can lock part of nodes as locked nodes. Sort all
+nodes by idle resource amount and select N nodes whose idle resource are the most. The N can be a fixed number or percentage.
+Make sure the potential available resource can satisfy the request of Big Job. Then check if the available resource meets
+the Big Job's demand after exist tasks finishing in locked nodes releases resource every scheduling cycle. This scheme is
+suitable for users who are very experienced with his/her usage scenarios and the job type is almost the same(the run-time
+gap of tasks is not too large).
+#### Single-Node Lock
+Another way to lock N nodes is lock one node every scheduling cycle. The selected node has the most idle resource at that
+cycle. The dynamic selection process can alleviate the stereotype caused by one-time selection, especially in scene that
+task type is unpredictable and complex.
+
+![Feature Design](./images/ReservationDesign.png)
+
+## Implementation
+### Action
+Add two new action: elect and reserve. Elect action aims to find the Big Job. Reserve action is responsible for select
```

Comment:
The reserve algorithm can be implemented as a plugin so that customized algorithms can also be added

Reply:
> The reserve algorithm can be implemented as a plugin so that customized algorithms can also be added

yes, current implementation does did it like this